core/src/test/scala/sttp/tapir/ShadowedEndpointCheckerTest.scala

```diff
@@ -0,0 +1,107 @@
+package sttp.tapir
+
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import sttp.tapir.util.ShadowedEndpointChecker
+
+class ShadowedEndpointCheckerTest extends AnyFlatSpecLike with Matchers {
+
+  it should "should detect shadowed endpoints with path variables that contain special characters" in {
+    val e1 = endpoint.get.in("[x]")
+    val e2 = endpoint.get.in("x")
+    val e3 = endpoint.get.in("/..." / paths)
+    val e4 = endpoint.get.in("x/")
+    val e5 = endpoint.get.in("x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3, e4, e5))
+
+    val expectedResult = List(ShadowedEndpoint(e5, e2))
+    result shouldBe expectedResult
+  }
+
+  it should "detect endpoint with wildcard shadowed by endpoint with fixed path" in {
+    val e1 = endpoint.get.in("x/y/z")
+    val e2 = endpoint.get.in("x" / "y" / "z")
+    val e3 = endpoint.get.in("x" / paths)
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3))
+
+    val expectedResult = List(ShadowedEndpoint(e3, e2))
+    result shouldBe expectedResult
+  }
+
+  it should "detect endpoint with path variable shadowed by endpoint with wildcard" in {
+    val e1 = endpoint.get.in("x" / paths)
+    val e2 = endpoint.get.in("x" / path[String].name("y_2") / "z")
+    val e3 = endpoint.get.in("x" / "y" / "x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3))
+
+    val expectedResult = List(ShadowedEndpoint(e2, e1), ShadowedEndpoint(e3, e1))
+    result shouldBe expectedResult
+  }
+
+  it should "detect shadowed endpoints for endpoints with similar path structure but with different path vars names" in {
+    val e1 = endpoint.get.in("x" / "y")
+    val e2 = endpoint.get.in("x" / "y" / paths)
+    val e3 = endpoint.get.in("x" / path[String].name("y_2") / path[String].name("y_4") / "z1")
+    val e4 = endpoint.get.in("x" / path[String].name("y_3") / path[String].name("y_5") / "z2")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3, e4))
+
+    val expectedResult = List(ShadowedEndpoint(e2, e1), ShadowedEndpoint(e3, e2), ShadowedEndpoint(e4, e2))
```

Comment:
same here - there are requests where e2 would be invoked, so it's not shadowed by e1. Only partially, but we're trying to detect total shadowing :)